modules/ximgproc/src/sparse_match_interpolators.cpp

```diff
@@ -877,5 +888,1148 @@ bool operator<(const SparseMatch& lhs,const SparseMatch& rhs)
         return (lhs.reference_image_pos.x<rhs.reference_image_pos.x);
 }
 
+class RICInterpolatorImpl CV_FINAL : public RICInterpolator
+{
+public:
+    static Ptr<RICInterpolatorImpl> create();
+    void interpolate(InputArray from_image, InputArray from_points, InputArray to_image, InputArray to_points, OutputArray dense_flow) CV_OVERRIDE;
+
+protected:
+    // internal buffers
+    int match_num;
+    vector<node>* g;
+    Mat NNlabels;
+    Mat NNdistances;
+    Mat labels;
+    Mat costMap;
+    static const int distance_transform_num_iter = 1;
+    float lambda;
+
+    //tunable parameters:
+    int max_neighbors;
+    float alpha;
+    int sp_size;
+    int sp_nncnt;
+    float sp_ruler;
+    int model_iter;
+    bool refine_models;
+    bool use_variational_refinement;
+    float cost_suffix;
+    float max_flow;
+    bool use_global_smoother_filter;
+    float fgs_lambda;
+    float fgs_sigma;
+    SLICType slic_type;
+
+    class GetKNNMatches_ParBody : public ParallelLoopBody
```

Comment:
BTW, "master" branch allows using of C++11 features, including lambdas for `parallel_for_()`.

It is not necessary to move all code into caller functions, you might call methods from lambdas (with extra parameters if needed).

Reply:
Can you give me a link to an function in the OpenCV lib where this has been used. So I can see how lambdas in that case are used correctly?

Reply:
done